src/hera/task.py

```diff
@@ -500,7 +502,7 @@ def deduce_parameters(
                             "`with_params` is empty and there exists non-default arguments "
                             f"which aren't covered by `inputs`: {missing_args}"
                         )
-            elif isinstance(self.with_param, Union[str, Parameter]):
+            elif isinstance(self.with_param, str) or isinstance(self.with_param, Parameter):
```

Comment:
Aren't these functionally equivalent? I feel like the former reads better.

Reply:
Does it work in all versions of Python 3.7+? 😮

Reply:
I tried the following in Hera's virtual environment and it raised an error:
```python
from typing import Union

class A:
    f: int = 42

def foo(x: Union[str, A]):
    print(isinstance(x, Union[str, A]))

foo(a)
```

```
~/.pyenv-x86_64/versions/3.7.12/lib/python3.7/typing.py in __subclasscheck__(self, cls)
    722             if cls._special:
    723                 return issubclass(cls.__origin__, self.__origin__)
--> 724         raise TypeError("Subscripted generics cannot be used with"
    725                         " class and instance checks")
    726 

TypeError: Subscripted generics cannot be used with class and instance checks

```

Reply:
These both run fine for me, but I'm on python 3.10. So yeah, let's use your revision :+1: